kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/RemoveFile.java

```diff
@@ -36,8 +42,95 @@ public class RemoveFile {
           .add("deletionVector", DeletionVectorDescriptor.READ_SCHEMA, true /* nullable */)
           .add("baseRowId", LongType.LONG, true /* nullable */)
           .add("defaultRowCommitVersion", LongType.LONG, true /* nullable */);
-  // TODO: Currently, Kernel doesn't create RemoveFile actions internally, nor provides APIs for
-  //  connectors to generate and commit them. Once we have the need for this, we should ensure
-  //  that the baseRowId and defaultRowCommitVersion fields of RemoveFile actions are correctly
-  //  populated to match the corresponding AddFile actions.
+  // TODO: Currently Kernel doesn't support RemoveFile actions when rowTracking is enabled (or have
+  //   any public API for generating RemoveFile actions). Once we
+  //   do this we need to ensure that the baseRowId and defaultRowCommitVersion fields are correctly
```

Comment:
does it make sense to have have these checks in the Transaction.commit? Separate PR is fine though if you think it is needed.

Reply:
Are you thinking throw exception if rowTracking is enabled and we see a remove? That seems like a reasonable thing to do at commit. I can do this as a follow-up